lib/trike/cloud_event.ex

```
@@ -0,0 +1,61 @@
+defmodule Trike.CloudEvent do
+  @moduledoc """
+  Represents a standard CloudEvent as well as a function for creating new
+  CloudEvents from OCS messages.
+  """
+
+  alias Trike.OcsRawMessage
+
+  @type t() :: %__MODULE__{
+          specversion: String.t(),
+          type: String.t(),
+          source: String.t(),
+          id: String.t(),
+          partitionkey: String.t(),
+          time: DateTime.t(),
+          data: OcsRawMessage.t()
+        }
+
+  @derive Jason.Encoder
+  @enforce_keys [:source, :id, :partitionkey, :time, :data]
+  defstruct @enforce_keys ++
+              [
+                specversion: "1.0",
+                type: "com.mbta.ocs.raw_message"
+              ]
+
+  @doc """
+  Creates a CloudEvent struct given a full OCS message, the current time, and a
+  partition key.
+  """
+  @spec from_ocs_message(binary(), DateTime.t(), String.t()) :: t()
```

I'm thinking this should maybe return `{:ok, t()}` or `:error`. It's *possible* the message is malformed, though unlikely. And while "let it crash" is often the Elixir philosophy, I'm worried in particular about getting wedged where we read a partial message, that causes this to crash, and then when it starts up again, it will read a partial message and crash again, and repeat.